proofs/tree.go

```diff
@@ -211,6 +211,20 @@ type Salt struct {
 type Salts []Salt
 type GetSalt func(compact []byte) []byte
 
+type PropParam struct {
```

Comment:
is this really necessary? Why not just pass in a string or a byte slice when needed?

Reply:
It's only used in one method, CreateProof. Given an empty value is invalid, you could just check which of the two values are len()> 0 or we could create two `CreateProof` methods?

Reply:
The reason to create above type was to help handle any logic in conversions. But the conversions wasn't needed for now.
I can create two methods, 
- CreateProofForReadableProp
- CreateProofForCompactProp
And wrap `CreateProof` method using those, feels nicer as an interface. What do you think?

Reply:
I would prefer to remove the struct as it shouldn't be needed. And yes adding the two public methods make sense.

Reply:
done